tmt/steps/execute/internal.py

```diff
@@ -283,10 +283,12 @@ def _handle_reboot(self, test: Test, guest: Guest) -> bool:
                 tmt.steps.execute.TEST_DATA)
             with open(reboot_request_path, 'r') as reboot_file:
                 reboot_data = json.loads(reboot_file.read())
-            try:
-                reboot_command = ShellScript(reboot_data.get('command'))
-            except TypeError:
-                reboot_command = None
+            reboot_command = None
+            if reboot_data.get('command'):
+                try:
+                    reboot_command = ShellScript(reboot_data.get('command'))
+                except TypeError:
+                    pass
```

Comment:
I don't understand what was wrong with the code. I *think* the `try/except` was there to catch situations where `get('command')` would return the default `None` - if `reboot_data.get('command')` not false-ish, `try/except` should not be needed.

An empty string as the reboot command in the log you mentioned, so, I guess `reboot_data.get('command')` may return an empty string? One more thing to fix, I suppose, `ShellCommand('')` should probably raise an exception... In a follow-up patch for 1.22, I don't wish to complicate your situation even more, filed https://github.com/teemtee/tmt/issues/1831.

Reply:
tmt-reboot when run without -c produces empty string as a command.